### PR TITLE
Calculates and plots repository code churn

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/repository_churn.py  6
- varats/utils/git_util.py  8
- varats/plots/blame_interaction_degree.py  3
         

Clones removed
==============
+ varats/plots/blame_interaction_degree.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VaRA/VaRA-Tool-Suite/pullRequest?prid=4736088&bid=15810180)

File: varats/plots/repository_churn.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        style.use(self.style)
+
+        case_study: CaseStudy = self.plot_kwargs['plot_case_study']
+
+        _, axis = plt.subplots()
+        draw_code_churn(
+            axis, self.plot_kwargs['project'], self.plot_kwargs['get_cmap'](),
+            case_study.has_revision if case_study else lambda x: True)
+        axis.set_yscale('symlog')
```
Comment (boehmseb) [inline varats/plots/repository_churn.py]: I would argue that log-scale is not always the best choice here. Can we make this optional?

File: varats/plots/repository_churn.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    Draws a churn plot onto an axis, showing insertions with green and
+    deletions with red.
+
+    Args:
+        revision_selector: takes a revision string and returns True if this rev
+                           should be included
+    """
+    code_churn = _build_repo_churn_table(project_name, commit_map)
+
+    # code_churn = cs_selector(code_churn)
```
Comment (boehmseb) [inline varats/plots/repository_churn.py]:
```suggestion
```

File: varats/plots/blame_interaction_degree.py [hunk 3]
```diff
@@ -217,21 +232,36 @@ def cs_filter(data_frame: pd.DataFrame) -> pd.DataFrame:
...
+
+            revs = [
+                x
+                for x in map(lambda x: tp.cast(object,
+                                               x.split('-')[1]), uniq_revisions)
+            ]
+            draw_code_churn(churn_axis, self.plot_kwargs['project'],
+                            self.plot_kwargs['get_cmap'](),
+                            lambda x: x[:10] in revs)
+            churn_axis.set_yscale('symlog')
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]: I would argue that log-scale is not always the best choice here. Can we make this optional?

File: varats/plots/blame_interaction_degree.py [hunk 4]
```diff
@@ -217,21 +232,36 @@ def cs_filter(data_frame: pd.DataFrame) -> pd.DataFrame:
...
+                                      'family': 'monospace'
+                                  })
         plt.setp(legend.get_title(),
                  fontsize=plot_cfg['legend_size'],
                  family='monospace')
 
-        for y_label in axis.get_yticklabels():
+        # draw churn subplot
+        if with_churn:
+            # for rev in uniq_revisions[::1]:
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]:
```suggestion
```

File: varats/plots/blame_interaction_degree.py [hunk 5]
```diff
@@ -217,21 +232,36 @@ def cs_filter(data_frame: pd.DataFrame) -> pd.DataFrame:
...
+            #     main_axis.axvline(rev, color='0.90', linewidth=1)
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]:
```suggestion
```

File: varats/plots/blame_interaction_degree.py [hunk 6]
```diff
@@ -217,21 +232,36 @@ def cs_filter(data_frame: pd.DataFrame) -> pd.DataFrame:
...
+
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]:
```suggestion
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/repository_churn.py  6
- varats/utils/git_util.py  8
- varats/plots/blame_interaction_degree.py  3
         

Clones removed
==============
+ varats/plots/blame_interaction_degree.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VaRA/VaRA-Tool-Suite/pullRequest?prid=4736088&bid=15810180)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/repository_churn.py  6
- varats/utils/git_util.py  8
- varats/plots/blame_interaction_degree.py  3
         

Clones removed
==============
+ varats/plots/blame_interaction_degree.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VaRA/VaRA-Tool-Suite/pullRequest?prid=4736088&bid=15810180)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/repository_churn.py  6
- varats/utils/git_util.py  8
- varats/plots/blame_interaction_degree.py  3
         

Clones removed
==============
+ varats/plots/blame_interaction_degree.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/VaRA/VaRA-Tool-Suite/pullRequest?prid=4736088&bid=15810180)

**Review by boehmseb — APPROVED** [review]
lgtm

**Pull request merged**